01-data-structures/05-hashes-part-2/separate_chaining/separate_chaining.rb

```diff
@@ -1,90 +1,116 @@
 require_relative 'linked_list'
+require_relative 'node'
 
 class SeparateChaining
   attr_reader :max_load_factor
 
   def initialize(size)
     @items = Array.new(size)
-    @item_count = 0
     @max_load_factor = 0.7
   end
 
   def []=(key, value)
-    i = index(key, @items.size)
-    n = Node.new(key, value)
+    index = self.index(key, self.size)
+    node = Node.new(key, value)
 
-    # COLLISION!
-    @items[i] != nil ? list = @items[i] : list = LinkedList.new
+    if @items[index] == nil
+      list = LinkedList.new()
+      list.add_to_front(node)
+      @items[index] = list
+    else 
+      @items[index].add_to_front(node) 
+    end
 
-    list.add_to_tail(n)
-    @items[i] = list
-    @item_count = @item_count + 1
+    self.resize() if load_factor > self.max_load_factor
+  end
 
-    # Resize the hash if the load factor grows too large
-    if load_factor.to_f > max_load_factor.to_f
-      resize
+  def [](key)
+    index = self.index(key, self.size)
+    if @items[index].head == nil 
```

Comment:
This definitely works, but you could simplify the code and make it a bit cleaner with the following:

```ruby
  index = self.index(key, self.size)
  return nil if @items[index] == nil
  node = @items[index].head
  while node != nil
      if node.key == key
        return node.value
      end
      node = node.next
  end
```